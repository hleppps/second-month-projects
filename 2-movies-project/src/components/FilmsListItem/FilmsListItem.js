import { Card } from "react-bootstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { useContext } from "react";
import AuthContext from "../../store/auth-context";
import styles from "./FilmsListItem.module.scss";
import * as CONSTANTS from "../../utils/constants";
import "../../utils/fontawasome";

const FilmsListItem = (props) => {
  const { isLoggedIn } = useContext(AuthContext);

  const iconType = props.isFavorite ? "fas" : "far";

  return (
    <>
      <Card className={styles.card}>
        {props.backdropPath && (
          <Card.Img
            variant="top"
            src={`${CONSTANTS.IMAGE_DOMAIN}${props.backdropPath}`}
            alt={props.backdropPath}
          />
        )}
        {!props.backdropPath && (
          <Card.Img variant="top" src={CONSTANTS.ERROR_IMAGE_URL} />
        )}
        <Card.Body className={styles.body}>
          <Card.Title data-testid="title">{props.title}</Card.Title>
          <Card.Text data-testid="subtitle">Rating: {props.rating}</Card.Text>
        </Card.Body>
        {isLoggedIn && (
          <Card.Footer>
            <FontAwesomeIcon
              data-testid="favorite-icon"
              icon={[iconType, "star"]}
              onClick={() => {
                props.setFavoriteFilm(props.filmId, props.isFavorite);
              }}
            />
          </Card.Footer>
        )}
      </Card>
    </>
  );
};

export default FilmsListItem;
