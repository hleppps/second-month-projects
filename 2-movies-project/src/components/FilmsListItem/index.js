import FilmsListItem from "./FilmsListItem";

export default FilmsListItem;
