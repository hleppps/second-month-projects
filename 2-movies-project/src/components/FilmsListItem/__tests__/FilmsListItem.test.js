import "@testing-library/jest-dom";
import { cleanup, render, fireEvent } from "@testing-library/react";
import AuthContext from "../../../store/auth-context";
import FilmsListItem from "../FilmsListItem";

const renderWithContext = (component, providerProps) => {
  return {
    ...render(
      <AuthContext.Provider value={providerProps}>
        {component}
      </AuthContext.Provider>
    ),
  };
};

afterEach(cleanup);

describe("FilmsListItems component", () => {
  it("renders with correct values from props", () => {
    const title = "Film Title";
    const rating = "7.8";
    const backdropPath = "/link.jpg";

    const { getByTestId, getByAltText } = render(
      <FilmsListItem
        title={title}
        rating={rating}
        backdropPath={backdropPath}
      />
    );

    expect(getByTestId("title")).toHaveTextContent(title);
    expect(getByTestId("subtitle")).toHaveTextContent(rating);
    getByAltText(backdropPath);
  });

  it("renders favorites when logged in", () => {
    const setFavoriteFilm = jest.fn();
    const isFavorite = false;
    const providerProps = {
      isLoggedIn: true,
    };

    const { getByTestId } = renderWithContext(
      <FilmsListItem
        setFavoriteFilm={setFavoriteFilm}
        isFavorite={isFavorite}
      />,
      providerProps
    );

    const favoriteIcon = getByTestId("favorite-icon");

    fireEvent.click(favoriteIcon);

    expect(setFavoriteFilm).toBeCalled();
  });

  it("matches a snapshot", () => {
    const { asFragment } = render(<FilmsListItem />);
    expect(asFragment(<FilmsListItem />)).toMatchSnapshot();
  });
});
