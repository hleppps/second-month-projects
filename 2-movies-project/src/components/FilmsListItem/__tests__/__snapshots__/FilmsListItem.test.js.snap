// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmsListItems component matches a snapshot 1`] = `
<DocumentFragment>
  <div
    class="card card"
  >
    <img
      class="card-img-top"
      src="https://i.imgur.com/GIeKF0n.png"
    />
    <div
      class="body card-body"
    >
      <div
        class="card-title h5"
        data-testid="title"
      />
      <p
        class="card-text"
        data-testid="subtitle"
      >
        Rating: 
      </p>
    </div>
  </div>
</DocumentFragment>
`;
